nimbus/src/enrollment.rs

```diff
@@ -149,36 +149,19 @@ impl ExperimentEnrollment {
                     updated_enrollment
                 }
             }
-            EnrollmentStatus::Enrolled {
-                ref branch,
-                enrollment_id,
-                ..
-            } => {
+            EnrollmentStatus::Enrolled { ref branch, .. } => {
                 if !is_user_participating {
                     log::debug!(
                         "Existing experiment enrollment '{}' is now disqualified (global opt-out)",
                         &self.slug
                     );
-                    let updated_enrollment = Self {
-                        slug: self.slug.clone(),
-                        status: EnrollmentStatus::Disqualified {
-                            reason: DisqualifiedReason::OptOut,
-                            enrollment_id,
-                            branch: branch.clone(),
-                        },
-                    };
+                    let updated_enrollment =
+                        self.disqualify_if_enrolled(DisqualifiedReason::OptOut);
```

Comment:
We do this disqualification pattern in enough places that it seemed like a helper method would be useful.